go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -247,13 +260,32 @@ func runOvnKube(ctx *cli.Context) error {
 				panic("Cannot initialize node without service account 'token'. Please provide one with --k8s-token argument")
 			}
 
-			err := clusterController.StartClusterNode(node)
+			err = clusterController.StartClusterNode(node)
 			if err != nil {
 				logrus.Errorf(err.Error())
 				panic(err.Error())
 			}
 		}
 	}
+
+	if clusterController.GatewayInit && clusterController.GatewayName != "" &&
+		node == "" {
```

Comment:
Invoked only when node=="" ?

Reply:
The StartClusterNode code calls InitGateway code. So calling it again would be a duplicate. But I see that the code is confusing and may be hard to maintain. It may make sense to get rid of InitGateway call from StartClusterNode. I will work on that.